modules/sql/managed-instance/vulnerability-assessment/main.bicep

```diff
@@ -39,12 +44,21 @@ resource managedInstance 'Microsoft.Sql/managedInstances@2022-02-01-preview' exi
   name: managedInstanceName
 }
 
+module roleAssignment '.bicep/nested_storageRoleAssignment.bicep' = if (!(useStorageAccountAccessKey)) {
+  name: guid(managedInstance.id, managedInstance.name, 'Storage Blob Data Contributor')
```

Comment:
```suggestion
  name: guid(managedInstance.id, managedInstance.entity.principalId, 'Storage Blob Data Contributor')
```

Reply:
Using the MI principal ID throws this error: This expression is being used in an assignment to the "name" property of the "module" type, which requires a value that can be calculated at the start of the deployment. Properties of managedInstance which can be calculated at the start include "apiVersion", "id", "name", "type"